### ISSUE TITLE
Legendary? Resuming installation cancels after 15 seconds, ui shows not installed. Importing fixed it.

### ISSUE THREAD

**Issue by StefanLobbenmeier** [Heroic-Games-Launcher/HeroicGamesLauncher#1659]
### Describe the bug

Not quite sure what caused this. In the log we can see a success notification (by legendary?), so I think legendary does not have an issue. 

I could resolve the issue by pressing on the import game button instead of install. I did several attempts of presssing "Install", but it always stopped after around 15 seconds.

I had previously stopped and resumed the download a few times in the last days, so it might be related to this.

### Add logs

(21:50:28) DEBUG:   [Legendary]:       Running Legendary command: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary --version

(21:50:37) INFO:    [Legendary]:       Legendary location: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary

(21:50:37) INFO:    [Gog]:             GOGDL location: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\gogdl

(21:50:37) INFO:    [Backend]:         

Heroic Version: 2.3.10 Brook

Legendary Version:  0.20.27 Dark Energy (hotfix)

OS: Microsoft Windows 10 Education KERNEL: 10.0.19044 ARCH: x64

CPU: Intel Core™ i5-8600K @3.6 

RAM: Total: 15.88 GiB Available: 9.2 GiB

GRAPHICS: GPU0: Intel(R) UHD Graphics 630 VRAM: 1024MB DRIVER:  GPU1: NVIDIA GeForce RTX 3070 VRAM: 8192MB DRIVER: 516.59 

(21:50:39) WARNING: [Backend]:         Protocol already registered.

(21:50:41) INFO:    [Frontend]:        Refreshing Library

(21:50:41) INFO:    [Legendary]:       Refreshing library...

(21:50:41) INFO:    [Legendary]:       Refreshing Epic Games...

(21:50:41) DEBUG:   [Legendary]:       Running Legendary command: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list

(21:50:45) INFO:    [Legendary]:       Updating game list

(21:50:45) INFO:    [Legendary]:       Game List Updated

(21:50:45) INFO:    [Legendary]:       Checking for game updates: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list-installed --check-updates --tsv

(21:50:45) DEBUG:   [Legendary]:       Running Legendary command: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary list-installed --check-updates --tsv

(21:50:47) INFO:    [Legendary]:       Found 0 game(s) to update

(21:50:47) INFO:    [Gog]:             Found 0 game(s) to update

(21:51:00) DEBUG:   [Legendary]:       Using cached install info

(21:51:00) DEBUG:   [Legendary]:       Using cached install info

(21:51:03) DEBUG:   [Legendary]:       Using cached install info

(21:51:03) INFO:    [Legendary]:       Installing 890d9cf396d04922a1559333df419fed with: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary install 890d9cf396d04922a1559333df419fed --platform Windows --base-path C:\Users\Stefan\Games\Heroic --skip-dlcs --skip-sdl -y

(21:51:03) DEBUG:   [Legendary]:       Running Legendary command: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary install 890d9cf396d04922a1559333df419fed --platform Windows --base-path C:\Users\Stefan\Games\Heroic --skip-dlcs --skip-sdl -y

(21:51:03) DEBUG:   [Legendary]:       Logging to file "C:\Users\Stefan\AppData\Roaming\heroic\GamesConfig\890d9cf396d04922a1559333df419fed.log"

(21:51:06) DEBUG:   [Legendary]:       Using cached install info

(21:51:07) INFO:    [Backend]:         finished installing

(21:51:07) INFO:    [Frontend]:        Refreshing Library

(21:51:07) INFO:    [Legendary]:       Refreshing library...

(21:51:07) INFO:    [Legendary]:       Updating game list

(21:51:07) INFO:    [Legendary]:       Game List Updated

(21:51:07) INFO:    [Backend]:         operation interrupted

(21:51:07) DEBUG:   [Legendary]:       Using cached install info

(21:51:07) DEBUG:   [Legendary]:       Using cached install info

(21:51:07) DEBUG:   [Legendary]:       Using cached install info

(21:51:09) DEBUG:   [Legendary]:       Using cached install info

(21:51:09) DEBUG:   [Legendary]:       Using cached install info

(21:51:14) INFO:    [Legendary]:       Importing 890d9cf396d04922a1559333df419fed: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary import 890d9cf396d04922a1559333df419fed C:\Users\Stefan\Games\Heroic\ShadowoftheTombRaider

(21:51:14) DEBUG:   [Legendary]:       Running Legendary command: C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary import 890d9cf396d04922a1559333df419fed C:\Users\Stefan\Games\Heroic\ShadowoftheTombRaider

(21:51:14) DEBUG:   [Legendary]:       Using cached install info

(21:51:17) INFO:    [Backend]:         imported Shadow of the Tomb Raider: Definitive Edition

(21:51:17) INFO:    [Frontend]:        Refreshing Library

(21:51:17) INFO:    [Legendary]:       Refreshing library...

(21:51:17) INFO:    [Legendary]:       Updating game list

(21:51:17) INFO:    [Legendary]:       Game List Updated

(21:51:18) DEBUG:   [Legendary]:       Using cached install info

(21:51:18) DEBUG:   [Legendary]:       Using cached install info

(21:51:18) DEBUG:   [Legendary]:       Using cached install info

### Steps to reproduce

Unsure if that works but try:

1. start a download, wait a bit
2. stop the download, restart pc, resume the download
3. 2 again, this time the download finishes
4. somehow the finishing was not registered? attempt to install again, but it never completes, just aborts

### Expected behavior

1. after the log message "Instaleld", the game shows as installed in the ui

### System Information

- OS [e. g. "Windows"]: 21H2
- Heroic Version [e. g. 2.1.1]:2.3.10 Brook

**Comment by StefanLobbenmeier** [comment]
Another idea on how reproducing might work:
1. clear legendary data, but dont uninstall the game
2. attempt to reinstall the game via heroic

Might not work, but worth a try if we can figure out how to clear legendary data

**Comment by Nocccer** [comment]
Can you try heroic beta? Maybe it is fixed there?

**Comment by arielj** [comment]
would be useful to see the output of manually running this when the installation is incorrect

```
C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary install 890d9cf396d04922a1559333df419fed --platform Windows --base-path C:\Users\Stefan\Games\Heroic --skip-dlcs --skip-sdl -y
```

Just to see if there's more info running legendary directly.

Also this might help:

```
C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info 890d9cf396d04922a1559333df419fed
```

**Comment by StefanLobbenmeier** [comment]
Thanks :) I’ll try to reproduce tonight

**Comment by StefanLobbenmeier** [comment]
Wasnt able to break it yet, but this is the output while it is installed (which explains the pause that I saw in the Heroic UI where it still said installing)

```
C:\Users\Stefan>C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary install 890d9cf396d04922a1559333df419fed --platform Windows --base-path C:\Users\Stefan\Games\Heroic --skip-dlcs --skip-sdl -y
[cli] INFO: Game needs to be verified before updating, switching to repair mode...
[Core] INFO: Logging in...
[Core] INFO: Fetching metadata for 1 app(s).
[cli] INFO: Game has not been verified yet.
[cli] INFO: Loading installed manifest for "890d9cf396d04922a1559333df419fed"
[cli] INFO: Verifying "Shadow of the Tomb Raider: Definitive Edition" version "TR11_09122021_458.0_1224786_740558_132835443930078066_"
Verification progress: 3/391 (0.0%) [0.0 MiB/s]
=> Verifying large file "bigfile.000.tiger": 100% (2047.8/2047.8 MiB) [448.0 MiB/s]
Verification progress: 4/391 (3.2%) [440.6 MiB/s]
=> Verifying large file "bigfile.001.tiger": 100% (2044.5/2044.5 MiB) [465.7 MiB/s]
Verification progress: 5/391 (6.4%) [465.6 MiB/s]
=> Verifying large file "bigfile.002.tiger": 100% (2047.7/2047.7 MiB) [468.3 MiB/s]
Verification progress: 6/391 (9.5%) [468.2 MiB/s]
=> Verifying large file "bigfile.003.tiger": 100% (2047.9/2047.9 MiB) [474.9 MiB/s]
Verification progress: 7/391 (12.7%) [474.8 MiB/s]
=> Verifying large file "bigfile.004.tiger": 100% (2046.5/2046.5 MiB) [476.5 MiB/s]
Verification progress: 8/391 (15.9%) [476.4 MiB/s]
=> Verifying large file "bigfile.005.tiger": 100% (2047.6/2047.6 MiB) [471.3 MiB/s]
Verification progress: 9/391 (19.1%) [471.2 MiB/s]
=> Verifying large file "bigfile.006.tiger": 100% (2047.9/2047.9 MiB) [481.8 MiB/s]
Verification progress: 10/391 (22.3%) [481.9 MiB/s]
=> Verifying large file "bigfile.007.tiger": 100% (2035.4/2035.4 MiB) [475.8 MiB/s]
Verification progress: 11/391 (25.4%) [475.9 MiB/s]
=> Verifying large file "bigfile.008.tiger": 100% (2048.0/2048.0 MiB) [486.5 MiB/s]
Verification progress: 12/391 (28.6%) [486.5 MiB/s]
=> Verifying large file "bigfile.009.tiger": 100% (2045.6/2045.6 MiB) [423.6 MiB/s]
Verification progress: 13/391 (31.8%) [423.8 MiB/s]
=> Verifying large file "bigfile.010.tiger": 100% (2048.0/2048.0 MiB) [427.0 MiB/s]
Verification progress: 14/391 (34.9%) [427.1 MiB/s]
=> Verifying large file "bigfile.011.tiger": 100% (1209.8/1209.8 MiB) [485.1 MiB/s]
Verification progress: 135/391 (37.0%) [484.9 MiB/s]
=> Verifying large file "bigfile.update1.002.000.tiger": 100% (2048.0/2048.0 MiB) [481.0 MiB/s]
Verification progress: 136/391 (40.2%) [426.7 MiB/s]
=> Verifying large file "bigfile.update1.002.001.tiger": 100% (1864.5/1864.5 MiB) [470.1 MiB/s]
Verification progress: 169/391 (45.9%) [460.0 MiB/s]
=> Verifying large file "bigfile.update2.002.000.tiger": 100% (2045.6/2045.6 MiB) [491.3 MiB/s]
Verification progress: 170/391 (49.1%) [463.1 MiB/s]
=> Verifying large file "bigfile.update2.002.001.tiger": 100% (754.2/754.2 MiB) [478.9 MiB/s]
Verification progress: 203/391 (52.4%) [361.8 MiB/s]
=> Verifying large file "bigfile.update3.002.000.tiger": 100% (2046.3/2046.3 MiB) [490.9 MiB/s]
Verification progress: 249/391 (56.7%) [261.0 MiB/s]
=> Verifying large file "bigfile_alltxt_arabic_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [489.1 MiB/s]
Verification progress: 251/391 (60.0%) [487.5 MiB/s]
=> Verifying large file "bigfile_alltxt_english_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [481.7 MiB/s]
Verification progress: 253/391 (63.3%) [479.6 MiB/s]
=> Verifying large file "bigfile_alltxt_french_allplt.000.tiger": 100% (2022.5/2022.5 MiB) [489.1 MiB/s]
Verification progress: 254/391 (66.4%) [487.6 MiB/s]
=> Verifying large file "bigfile_alltxt_german_allplt.000.tiger": 100% (2004.7/2004.7 MiB) [485.6 MiB/s]
Verification progress: 255/391 (69.5%) [485.4 MiB/s]
=> Verifying large file "bigfile_alltxt_iberspanish_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [472.6 MiB/s]
Verification progress: 257/391 (72.8%) [472.5 MiB/s]
=> Verifying large file "bigfile_alltxt_italian_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [488.6 MiB/s]
Verification progress: 259/391 (76.0%) [487.6 MiB/s]
=> Verifying large file "bigfile_alltxt_japanese_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [489.0 MiB/s]
Verification progress: 261/391 (79.3%) [485.8 MiB/s]
=> Verifying large file "bigfile_alltxt_ksa_english_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [490.3 MiB/s]
Verification progress: 263/391 (82.5%) [488.7 MiB/s]
=> Verifying large file "bigfile_alltxt_latamspanish_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [472.3 MiB/s]
Verification progress: 265/391 (85.8%) [470.9 MiB/s]
=> Verifying large file "bigfile_alltxt_polish_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [488.7 MiB/s]
Verification progress: 267/391 (89.1%) [486.2 MiB/s]
=> Verifying large file "bigfile_alltxt_portuguese_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [482.2 MiB/s]
Verification progress: 269/391 (92.4%) [480.8 MiB/s]
=> Verifying large file "bigfile_alltxt_russian_allplt.000.tiger": 100% (2048.0/2048.0 MiB) [484.2 MiB/s]
Verification progress: 271/391 (95.7%) [482.3 MiB/s]
=> Verifying large file "bigfile_alltxt_simplechinese_allplt.000.tiger": 100% (1987.7/1987.7 MiB) [485.9 MiB/s]
Verification progress: 391/391 (100.0%) [297.5 MiB/s]
[cli] INFO: Verification finished successfully.
[cli] INFO: Preparing download for "Shadow of the Tomb Raider: Definitive Edition" (890d9cf396d04922a1559333df419fed)...
[Core] INFO: Parsing game manifest...
[Core] INFO: Install path: C:\Users\Stefan\Games\Heroic\ShadowoftheTombRaider
[Core] INFO: Selected CDN: fastly-download.epicgames.com (https)
[DLM] INFO: Found previously interrupted download. Download will be resumed if possible.
[DLM] INFO: Skipping 391 files based on resume data.
[cli] INFO: Download size is 0, the game is either already up to date or has not changed. Exiting...

C:\Users\Stefan>
```

and info:
```
C:\Users\Stefan>C:\Users\Stefan\AppData\Local\Programs\heroic\resources\app.asar.unpacked\build\bin\win32\legendary info 890d9cf396d04922a1559333df419fed
[Core] INFO: Trying to re-use existing login session...

Game Information:
- App name: 890d9cf396d04922a1559333df419fed
- Title: Shadow of the Tomb Raider: Definitive Edition
- Latest version: TR11_09122021_458.0_1224786_740558_132835443930078066_
- All versions:
 +  Windows : TR11_09122021_458.0_1224786_740558_132835443930078066_
- Cloud saves supported: False
- Cloud save folder (Windows): (None)
- Cloud save folder (Mac): (None)
- Is DLC: False
- Activates on external platform: No
- Extra launch options: (None)
- Owned DLC: (None)

Installation information:
- Platform: Windows
- Version: TR11_09122021_458.0_1224786_740558_132835443930078066_
- Install size: 62.90 GiB
- Install path: C:\Users\Stefan\Games\Heroic\ShadowoftheTombRaider
- Save data path: (None)
- EGL sync GUID:
- Install tags: (None, all game data selected for install)
- Requires ownership verification token (DRM): False
- Installed DLC: (None)

Manifest information:
- Manifest size: 4072.2 KiB
- Manifest type: Binary
- Manifest version: 17
- Manifest feature level: 17
- Manifest app name: 1e99ed8ea4064da5945b6b0fdad4df0f
- Launch EXE: SOTTR.exe
- Launch Command: (None)
- Build version: TR11_09122021_458.0_1224786_740558_132835443930078066_
- Build ID: y9rTSjjgV9wz18dwvmkhbEPUrk8
- Prerequisites: (None)
- Install tags: (empty)
- Files: 391
- Chunks: 64404
- Disk size (uncompressed): 62.90 GiB
- Download size (compressed): 61.16 GiB
- Disk size by install tag: N/A
- Download size by install tag: N/A

C:\Users\Stefan>
```

**Comment by StefanLobbenmeier** [comment]
Ok I tried reproducing by deleting ~/.config/legendary/installed.json, which caused some very weird behaviour of legendary that it attempts to install everything again, but it still worked as in it downloaded and reported progress correctly.

I also tried to reproduce with a fresh game (a short hike), since it is much smaller so i can test better. I cancelled the download at 50%, then restarted my pc. Heroic correctly showed that the download was at 50% and offered to resume, which worked.
 I tried to cancel at the very end but after 100% downlaod it was also immediately installed.


So my best guess is that somehow for the larger game it has some longer process at the end and I interrupted that at a very unlucky moment in time. Hard to reproduce, sorry. So I think we should just keep it in the back of our minds for now, when I see the issue again I can reopen this issue. Maybe someone else will find a way to reproduce in the future either, but for now I am happy with closing this as non-reproducible.
